rero_ils/modules/notifications/dispatcher.py

```diff
@@ -47,39 +49,40 @@ def dispatch_notifications(cls, notification_pids=None, resend=False,
         :param verbose: Verbose output.
         :returns: dictionary with processed and send count
         """
-
         def get_dispatcher_function(channel):
             try:
                 communication_switcher = current_app.config.get(
                     'RERO_ILS_COMMUNICATION_DISPATCHER_FUNCTIONS', [])
                 return communication_switcher[channel]
             except KeyError:
                 current_app.logger.warning(
-                    'The communication channel of the patron'
-                    f' (pid: {patron["pid"]}) is not yet implemented')
+                    f'The communication channel: {channel}'
+                    ' is not yet implemented')
                 return Dispatcher.not_yet_implemented
 
         sent = not_sent = 0
         aggregated = {}
         pids = notification_pids if notification_pids else []
         notifications = [Notification.get_record_by_pid(pid) for pid in pids]
         notifications = list(filter(None, notifications))
-
         # BUILD AGGREGATED NOTIFICATIONS
         #   Notifications should be aggregated on (in this order):
         #     1. notification_type,
         #     2. communication_channel
         #     3. library
         #     4. patron
+        errors = 0
         for notification in notifications:
             try:
                 cls._process_notification(
                     notification, resend, aggregated)
             except Exception as error:
+                errors += 1
                 current_app.logger.error(
                     f'Notification has not be sent (pid: {notification.pid},'
                     f' type: {notification["notification_type"]}): '
                     f'{error}')
+                traceback.print_exc()
```

Comment:
I don't know where we can see the tracback print. Probably stderr, do we have to use the logger for this?

Reply:
maybe we ca use `exc_info=True` and/or `stack_info=True`  arguments in logger.error like:
```
current_app.logger.error(
                    f'Notification has not be sent (pid: {notification.pid},'
                    f' type: {notification["notification_type"]}): '
                    f'{error}', exc_info=True, stack_info=True)
```
see https://docs.python.org/3/library/logging.html#logging.Logger.debug

Reply:
Thanks, I learnt something.